### PR TITLE
IBX-1549: changed validation content to only one language

### PR DESCRIPTION
**Pull request by mateuszdebinski**
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | [IBX-1549](https://issues.ibexa.co/browse/IBX-1549), [IBX-1741](https://issues.ibexa.co/browse/IBX-1741) and [IBX-175](https://issues.ibexa.co/browse/IBX-175)
| **Type**                                   | bug
| **Target Ibexa version** | `v3.3`
| **BC breaks**                          | no

#### Checklist:
- [x] Provided PR description.
- [x] Tested the solution manually.
- [ ] Provided automated test coverage.
- [x] Checked that target branch is set correctly (master for features, the oldest supported for bugs).
- [x] Ran PHP CS Fixer for new PHP code (use `$ composer fix-cs`).
- [x] Asked for a review (ping `@ezsystems/engineering-team`).

### CODE REVIEW

**Review by Steveb-p — APPROVED** [review]

**Review by alongosz — CHANGES_REQUESTED** [review]
This change contains potential regression, which is hidden by `ContentUpdateStructValidator` and `ContentCreateStructValidator` because the validation happens also there and there's no easy way to manipulate content after `updateContent($draft)` but before `publishVersion($draft->versionInfo)`.

In other words, skipping validation here makes the Validator incoherent, looking from its independent and self-contained perspective. The business logic of API allows modifying multiple translations in a single update, so all translations should be valid. That said, I didn't manage to find a use case which would clearly uncover the bug introduced, because of the existence of other validators.

What I could agree on is for this Validator to validate only modified languages, determined by the `$translations` parameter of `\eZ\Publish\Core\Repository\ContentService::publishVersion`. If publish action occurred for all translations, it should behave as prior the change in this PR. Otherwise it should validate modified language(s) passed by `$translation`.

**Review by alongosz — APPROVED** [review]

**Review by Steveb-p — APPROVED** [review]

File: eZ/Publish/Core/Repository/Validator/VersionValidator.php [hunk 1]
```diff
@@ -68,7 +68,7 @@ public function validate(
             );
 
             foreach ($languageCodes as $languageCode) {
-                $fieldValue = $content->getField($fieldDefinition->identifier)->value ?? $fieldDefinition->defaultValue;
+                $fieldValue = $content->getField($fieldDefinition->identifier, $languageCode)->value ?? $fieldDefinition->defaultValue;
```
Comment (konradoboza) [inline eZ/Publish/Core/Repository/Validator/VersionValidator.php:71]: Unfortunately, this change would introduce a regression, ref: https://github.com/ezsystems/ezplatform-kernel/pull/246.

**Review by konradoboza — CHANGES_REQUESTED** [review]

File: eZ/Publish/Core/Repository/Validator/VersionValidator.php (hunk 1 above)
Comment (konradoboza) [reply eZ/Publish/Core/Repository/Validator/VersionValidator.php:71]: Well, to be more precise: if the above steps don't lead to the issue anymore concerning the scope of this PR, we should be good to go.

Comment (mateuszdebinski) [reply eZ/Publish/Core/Repository/Validator/VersionValidator.php:71]: I tested this ticket with steps from your comment, and the problem does not occur anymore

**Comment by mateuszdebinski** [comment]
@bogusez new ticket to test (with steps from comment): https://issues.ibexa.co/browse/IBX-175

**Review by konradoboza — APPROVED** [review]

**Review by alongosz — APPROVED** [review]

**Comment by bogusez** [comment]
> @bogusez new ticket to test (with steps from comment): https://issues.ibexa.co/browse/IBX-175

Already tested the scenario using changes from this PR and no exception occurs.

**Review by bogusez — APPROVED** [review]

**Pull request merged**